src/core/mac/mac.cpp

```diff
@@ -1142,6 +1142,7 @@ void Mac::ReceiveDoneTask(Frame *aFrame, ThreadError aError)
 
     VerifyOrExit(error == kThreadError_None, ;);
     VerifyOrExit(aFrame != NULL, error = kThreadError_NoFrameReceived);
+    SuccessOrExit(error = aFrame->ValidatePsdu());
```

Comment:
Suggestion: Moving this check after pcap callback.  If we get an invalid frame, it may still be valuable to see it in the pcap...Thoughts?